metadata_backend/api/handlers/object.py

```diff
@@ -139,28 +155,35 @@ async def post_object(self, req: Request) -> Response:
         data: Union[List[Dict[str, str]], Dict[str, str]]
         if isinstance(content, List):
             LOG.debug(f"Inserting multiple objects for {schema_type}.")
-            ids: List[Dict[str, str]] = []
+            objects: List[Dict[str, Any]] = []
             for item in content:
-                accession_id, title = await operator.create_metadata_object(collection, item[0])
-                ids.append({"accessionId": accession_id, "title": title})
-                LOG.info(f"POST object with accesssion ID {accession_id} in schema {collection} was successful.")
+                json_data = await operator.create_metadata_object(collection, item[0])
+                objects.append(json_data)
+                LOG.info(
+                    f"POST object with accesssion ID {json_data['accessionId']} in schema {collection} was successful."
+                )
+
             # we format like this to make it consistent with the response from /submit endpoint
-            data = [dict({"accessionId": item["accessionId"]}, **{"schema": schema_type}) for item in ids]
+            data = [dict({"accessionId": item["accessionId"]}, **{"schema": schema_type}) for item in objects]
             # we take the first result if we get multiple
             location_headers = CIMultiDict(Location=f"{url}/{data[0]['accessionId']}")
         else:
-            accession_id, title = await operator.create_metadata_object(collection, content)
-            data = {"accessionId": accession_id}
-            location_headers = CIMultiDict(Location=f"{url}/{accession_id}")
-            LOG.info(f"POST object with accesssion ID {accession_id} in schema {collection} was successful.")
+            json_data = await operator.create_metadata_object(collection, content)
+            data = {"accessionId": json_data["accessionId"]}
+            location_headers = CIMultiDict(Location=f"{url}/{json_data['accessionId']}")
+            LOG.info(
+                f"POST object with accesssion ID {json_data['accessionId']} in schema {collection} was successful."
+            )
+            objects = [json_data]
 
         # Gathering data for object to be added to folder
-        if not isinstance(data, List):
-            ids = [dict(data, **{"title": title})]
-        folder_op = FolderOperator(db_client)
-        patch = await self.prepare_folder_patch_new_object(collection, ids, patch_params)
+        patch = self._prepare_folder_patch_new_object(collection, objects, patch_params)
         await folder_op.update_folder(folder_id, patch)
 
+        # Create draft dataset to Metax catalog
+        if collection in _allowed_doi:
+            [await self._create_metax_dataset(req, collection, item, folder_id) for item in objects]
```

Comment:
is this a silent process that doesn't need to be returned anywhere? the list-comprehension confused me

Reply:
This is done as a shortcut to handle mypy errors. Changing to 
```
for item in objects:
    await self._create_metax_dataset(req, collection, item, folder_id)
``` 
produces mypy errors
```
Incompatible types in assignment (expression has type "Dict[str, Any]", variable has type "Tuple[Any, str]")
Argument 3 to "_create_metax_dataset" of "ObjectAPIHandler" has incompatible type "Tuple[Any, str]"; expected "Dict[Any, Any]"
```
which was a struggle to address

Reply:
ok :+1: